### PR TITLE
tests and modernizing infrastructure

### PR DESCRIPTION
**Pull request by leider**
Hi,

hope you like that. Feedback welcome. :)

### CODE REVIEW

**Comment by dhughes** [comment]
Oh awesome! I'll give this a look over tomorrow, time permitting. Feel free to harass me if it's been a few days. I should add that I haven't worked with Node.js in quite a long time. My ability to judge what is modern in this space. 

And tests! What a novel idea!

File: README.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+Current Build Status
+--------------------
+
+[![Build Status](https://travis-ci.org/leider/CoolBeans.png)](https://travis-ci.org/leider/CoolBeans)
```
Comment (dhughes) [inline README.md]: I've not used Travis before. I see that this references your fork. Should this be updated to reference my repo?

File: sample-config/dev.json [hunk 2]
```diff
@@ -1,180 +1,78 @@
...
-			{"value": ["localhost", "supporting.us", "qb.alagad.com:8433"]},
-			{"bean": "recipientValidator"},
-			{"bean": "codeValidator"},
-			{"bean": "creditCardValidator"},
-			{"bean": "userValidator"}
-		]
-	}
-}
-
+  "xmlbuilder": {"module": "xmlbuilder"},
```
Comment (dhughes) [inline sample-config/dev.json]: I'm honestly not a huge fan of compressing the sample JSON down like this. Can you justify why you made the change? I can see changing to 2 space tabbing (not my preference, but widely used), but smashing the more complicated configs like "service" and "creditCardDao".

Comment (leider) [inline sample-config/dev.json]: I'd prefer 2 space indentation. JSON formatting can also reverted. :)

File: README.md (hunk 1 above)
Comment (leider) [inline README.md]: Exactly. Was just for fun. :)

File: sample-config/dev.json (hunk 2 above)
Comment (leider) [inline sample-config/dev.json]: Also, I'd prefer a much simpler example without redundancy. But let's keep that for the future.

File: README.md (hunk 1 above)
Comment (dhughes) [inline README.md]: I just looked again and see that this doesn't reference anything at Github. I imagine your Travis-CL config references your fork? I'd think that we can keep this Travis setup but just point to my main repo.

File: sample-config/dev.json (hunk 2 above)
Comment (dhughes) [inline sample-config/dev.json]: Can you please take care of the formatting here and update the pull request? I'll then merge it in. 

Thanks for doing all this!

Comment (leider) [inline sample-config/dev.json]: Should I also change the spacing from tabs to spaces and reformat the js files?

Comment (dhughes) [inline sample-config/dev.json]: The sample file was from a defunct project I created CoolBeans for. If anyone has a more general example demonstrating all/most of the configuration options I think that'd be fine to use.

Comment (leider) [inline sample-config/dev.json]: I’ll strip it down.

Comment (dhughes) [inline sample-config/dev.json]: I'm fine with the two spaces. I just think some of these would be easier to read if broken into multiple lines as I had. For a sample, it just seems easier to understand.

**Comment by dhughes** [comment]
Sorry to take so long merging this in. It just fell off my radar!

**Pull request merged**